ocagent.go

```diff
@@ -125,7 +126,17 @@ func NewUnstartedExporter(opts ...ExporterOption) (*Exporter, error) {
 	viewDataBundler.BundleCountThreshold = 500 // TODO: (@odeke-em) make this configurable.
 	e.viewDataBundler = viewDataBundler
 	e.nodeInfo = NodeWithStartTime(e.serviceName)
-	e.resource = resourceProtoFromEnv()
+	if e.resourceDetector != nil {
+		res, err := e.resourceDetector(context.Background())
+		if err != nil {
+			panic(fmt.Sprintf("Error detecting resource. err:%v\n", err))
```

Comment:
Do we want to panic here? Or is it OK to just log a message and continue?

Reply:
yes for panic. for the same reasons as for vmmetricreceiver [comment](https://github.com/open-telemetry/opentelemetry-service/pull/31#pullrequestreview-252602891)